evap/static/js/datagrid.js

```diff
@@ -0,0 +1,323 @@
+                    orderValues,
+                };
+            });
+    }
+
+    searchWordsOf(string) {
+        return string.toLowerCase().trim().split(/\s+/);
+    }
+
+    // Filters rows respecting the current search string and filters by their searchWords and filterValues
+    filterRows() {
+        const searchWords = this.searchWordsOf(this.state.search);
+        for (const row of this.rows) {
+            const isDisplayedBySearch = searchWords.every(searchWord => {
+                return row.searchWords.some(rowWord => rowWord.includes(searchWord));
+            });
+            const isDisplayedByFilters = [...this.state.filter].every(([name, filterValues]) => {
+                return filterValues.some(filterValue => {
+                    return row.filterValues.get(name).some(rowValue => rowValue === filterValue);
+                });
+            });
+            row.isDisplayed = isDisplayedBySearch && isDisplayedByFilters;
+        }
+    }
+
+    sort(order) {
+        this.state.order = order;
+        this.sortRows();
+        this.renderToDOM();
+    }
+
+    // Sorts rows respecting the current order by their orderValues
+    sortRows() {
+        for (const header of this.sortableHeaders.values()) {
+            header.removeClass("col-order-asc col-order-desc");
+        }
+        for (const [column, ordering] of this.state.order) {
+            this.sortableHeaders.get(column).addClass(`col-order-${ordering}`);
+        }
+
+        this.rows.sort((a, b) => {
+            for (const [column, order] of this.state.order) {
+                if (a.orderValues.get(column) < b.orderValues.get(column)) {
+                    return order === "asc" ? -1 : 1;
+                } else if (a.orderValues.get(column) > b.orderValues.get(column)) {
+                    return order === "asc" ? 1 : -1;
+                }
+            }
+            return 0;
+        });
+    }
+
+    // Reflects changes to the rows to the DOM
+    renderToDOM() {
+        this.container.children().detach();
+        const elements = this.rows
+            .filter(row => row.isDisplayed)
+            .map(row => row.element);
+        this.container.append($(elements));
+        this.saveStateToStorage();
+    }
+
+    restoreStateFromStorage() {
+        const stored = JSON.parse(localStorage.getItem(this.storageKey)) || {};
+        this.state = {
+            search: stored.search || "",
+            filter: new Map(stored.filter),
+            order: stored.order || this.defaultOrder,
+        };
+        this.syncFilterStateWithInputs();
+        this.filterRows();
+        this.sortRows();
+        this.renderToDOM();
+    }
+
+    saveStateToStorage() {
+        const stored = {
+            search: this.state.search,
+            filter: [...this.state.filter],
+            order: this.state.order,
+        };
+        localStorage.setItem(this.storageKey, JSON.stringify(stored));
+    }
+
+    syncFilterStateWithInputs() {
```

Comment:
you're changing the inputs to match the state, right? I'd suggest to reflect that in the name somehow. `sync` goes both ways for me, or if it should be one way, you're syncing the inputs, not the state. at least I would say so :)